### PR TITLE
Simplify if/else statements in Types

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: libsolidity/ast/Types.cpp [hunk 1]
```diff
@@ -2354,14 +2355,26 @@ unsigned FunctionType::sizeOnStack() const
...
+		break;
+	case Kind::BareCall:
+	case Kind::BareCallCode:
+	case Kind::BareDelegateCall:
+	case Kind::Internal:
+	case Kind::ArrayPush:
+	case Kind::ByteArrayPush:
 		size = 1;
+		break;
+	default:
```
Comment (chriseth) [inline libsolidity/ast/Types.cpp:2374]: If possible, there should be no `default` case, so that we get a compiler warning whenever a new case is added.

**Review by chriseth — APPROVED** [review]

File: libsolidity/ast/Types.cpp (hunk 1 above)
Comment (axic) [reply libsolidity/ast/Types.cpp:2374]: It doesn't do that before the change. Also if the default is removed the compiler will complain that not all cases are handled.

Comment (chriseth) [reply libsolidity/ast/Types.cpp:2374]: yep, just "if possible".

Comment (axic) [reply libsolidity/ast/Types.cpp:2374]: We could add an assert? :)

**Pull request merged**